authors/apps/authentication/serializers.py

```diff
@@ -168,3 +171,49 @@ def update(self, instance, validated_data):
         instance.save()
 
         return instance
+
+
+class FacebookAuthSerializer(serializers.Serializer):
+    """ Performs facebook authorisation"""
+
+    access_token = serializers.CharField()
+
+    def validate_access_token(self, access_token):
+        """
+        This method validates the token received
+        :param access_token:
+        :return: auth_token
+        """
+        user_data = FacebookAuthentication.validate_token(access_token)
+
+        try:
+            user_data['id']
+        except:
+            msg = "Invalid or expired token"
+            raise serializers.ValidationError(msg)
+
+        return create_user_and_return_token(
+            user_data['id'], user_data['name'], user_data['email']
+            )
```

Comment:
Please ask @MuhanguziDavid to help you understad how to align your code and parens right.